src/content/docs/en/reference/api-reference.mdx

```diff
@@ -1120,14 +1189,16 @@ This means collections **cannot** store a mix of content and data formats. You m
 
 #### `schema`
 
-**Type:** `TSchema extends ZodType`
+**Type:** `TSchema extends ZodType | (context: SchemaContext) => TSchema extends ZodType`
```

Comment:
I wasn't sure how to update this type. Is `TSchema extends ZodType` really meaningful for someone new to Astro (or Typescript)? Reading this, it seems weird to me since the other types are usable while here, it is an indication for those who understand Typescript. Even if it's not a real type, shouldn't we just stick to `ZodType`?

```
`ZodType | (context: SchemaContext) => ZodType`
```

Reply:
Yes, I think your suggestion makes sense!

By the way, not required, but something we’ve done in the Starlight docs is actually link type definitions when we have them, i.e. here it might be:

```suggestion
**Type:** <code>ZodType | (context: <a href="#schemacontext">SchemaContext</a>) => ZodType</code>
```

Reply:
Nice idea, this can be useful to better understand the return type! I'll check the other types (I'm thinking in particular of `AstroCookie` ).